File: benchmark/output/python-pygame/openai/particle_gravity_well.py
import pygame
import sys
import random
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Gravity Well")
clock = pygame.time.Clock()

# Particle properties
px = []
py = []
pvx = []
pvy = []

# Gravity well properties
well_x = 400.0
well_y = 300.0
well_strength = 5000.0

def load():
    global px, py, pvx, pvy
    for _ in range(50):
        px.append(random.uniform(0.0, 800.0))
        py.append(random.uniform(0.0, 600.0))
        pvx.append(random.uniform(-30.0, 30.0))
        pvy.append(random.uniform(-30.0, 30.0))

def mousepressed(mx, my, button):
    global well_x, well_y
    well_x = mx
    well_y = my

def update(dt):
    global px, py, pvx, pvy
    for i in range(50):
        dx = well_x - px[i]
        dy = well_y - py[i]
        dist = math.sqrt(dx * dx + dy * dy)
        if dist > 5.0:
            force = well_strength / (dist * dist)
            pvx[i] += dx / dist * force * dt
            pvy[i] += dy / dist * force * dt

        px[i] += pvx[i] * dt
        py[i] += pvy[i] * dt

        if px[i] < 0.0:
            px[i] = 800.0
        if px[i] > 800.0:
            px[i] = 0.0
        if py[i] < 0.0:
            py[i] = 600.0
        if py[i] > 600.0:
            py[i] = 0.0

def draw():
    screen.fill((0, 0, 0))
    pygame.draw.circle(screen, (255, 255, 255), (int(well_x), int(well_y)), 8)
    for i in range(50):
        pygame.draw.rect(screen, (255, 255, 255), (int(px[i]), int(py[i]), 3, 3))
    font = pygame.font.Font(None, 36)
    text = font.render("Particles: 50", True, (255, 255, 255))
    screen.blit(text, (10, 10))
    text = font.render("Click to move well", True, (255, 255, 255))
    screen.blit(text, (10, 30))
    pygame.display.flip()

load()
running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mx, my = pygame.mouse.get_pos()
            mousepressed(mx, my, event.button)

    update(dt)
    draw()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/openai/particle_gravity_well.lua
local px = {}
local py = {}
local pvx = {}
local pvy = {}
local well_x = 400.0
local well_y = 300.0
local well_strength = 5000.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Gravity Well with Particles")
    for i = 1, 50 do
        px[i] = love.math.random(0, 800)
        py[i] = love.math.random(0, 600)
        pvx[i] = love.math.random(-30, 30)
        pvy[i] = love.math.random(-30, 30)
    end
end

function love.mousepressed(mx, my, button)
    if button == 1 then
        well_x = mx
        well_y = my
    end
end

function love.update(dt)
    for i = 1, 50 do
        local dx = well_x - px[i]
        local dy = well_y - py[i]
        local dist = math.sqrt(dx * dx + dy * dy)
        if dist > 5.0 then
            local force = well_strength / (dist * dist)
            pvx[i] = pvx[i] + (dx / dist) * force * dt
            pvy[i] = pvy[i] + (dy / dist) * force * dt
        end
        px[i] = px[i] + pvx[i] * dt
        py[i] = py[i] + pvy[i] * dt

        if px[i] < 0 then px[i] = 800 end
        if px[i] > 800 then px[i] = 0 end
        if py[i] < 0 then py[i] = 600 end
        if py[i] > 600 then py[i] = 0 end
    end
end

function love.draw()
    love.graphics.setColor(1, 1, 1)
    love.graphics.circle("fill", well_x, well_y, 8)
    for i = 1, 50 do
        love.graphics.rectangle("fill", px[i], py[i], 3, 3)
    end
    love.graphics.print("Particles: 50", 10, 10)
    love.graphics.print("Click to move well", 10, 30)
end